backend/tests/integration/files/test_api_mixin.py

```diff
@@ -0,0 +1,382 @@
+):
+    expected_document_key = "f276b50c9e6401b5e212785a496efa4e"
+    create_s3_bucket(expected_document_key)
+    data = {
+        "documents": [
+            {
+                "source": "arxiv",
+                "key": "key",
+                "url": "http://inspirehep.net/record/863300/files/fermilab-pub-10-255-e.pdf",
+                "original_url": "http://original-url.com/2",
+                "filename": "file1.pdf",
+            }
+        ]
+    }
+    create_record("lit", data=data)
+    assert s3.file_exists(expected_document_key) is True
+    metadata_document = s3.get_file_metadata(expected_document_key)
+    assert (
+        metadata_document["ContentDisposition"] == f'attachment; filename="file1.pdf"'
+    )
+    data["documents"][0]["filename"] = "file2.pdf"
+    create_record("lit", data=data)
+    assert s3.file_exists(expected_document_key) is True
+    metadata_document = s3.get_file_metadata(expected_document_key)
+    assert (
+        metadata_document["ContentDisposition"] == f'attachment; filename="file2.pdf"'
+    )
+
+
+@pytest.mark.vcr()
+def test_adding_record_with_documents_without_original_url(
+    base_app, db, es, create_record, enable_files, s3, create_s3_bucket
+):
+    expected_document_key = "f276b50c9e6401b5e212785a496efa4e"
+    create_s3_bucket(expected_document_key)
+    data = {
+        "documents": [
+            {
+                "source": "arxiv",
+                "key": "key",
+                "url": "http://inspirehep.net/record/863300/files/fermilab-pub-10-255-e.pdf",
+                "filename": "file1.pdf",
+            }
+        ]
+    }
+    record = create_record("lit", data=data)
+    expected_documents = [
+        {
+            "source": "arxiv",
+            "key": expected_document_key,
+            "url": f"{base_app.config.get('S3_HOSTNAME')}/{current_s3_instance.get_bucket(expected_document_key)}/{expected_document_key}",
+            "original_url": "http://inspirehep.net/record/863300/files/fermilab-pub-10-255-e.pdf",
+            "filename": "file1.pdf",
+        }
+    ]
+    assert s3.file_exists(expected_document_key) is True
+    assert expected_documents == record["documents"]
+
+
+@pytest.mark.vcr()
+def test_adding_deleted_record_with_documents_does_not_add_files(
+    base_app, db, es, create_record, enable_files, s3, create_s3_bucket
+):
+    expected_document_key = "f276b50c9e6401b5e212785a496efa4e"
+    create_s3_bucket(expected_document_key)
+    data = {
+        "deleted": True,
+        "documents": [
+            {
+                "source": "arxiv",
+                "key": "key",
+                "url": "http://inspirehep.net/record/863300/files/fermilab-pub-10-255-e.pdf",
+                "filename": "file1.pdf",
+            }
+        ],
+    }
+    create_record("lit", data=data)
+    assert s3.file_exists(expected_document_key) is False
+
+
+@pytest.mark.vcr()
+def test_update_record_with_documents_and_figures(
```

Comment:
what about update which removes some of the documents and figures and an update that adds more?